src/test/java/io/cdap/plugin/splunk/common/client/SplunkSearchIteratorTest.java

```diff
@@ -0,0 +1,126 @@
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Tests for {@link SplunkSearchIterator}
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(ResultsReader.class)
+@PowerMockIgnore({"javax.net.ssl.*"})
+public class SplunkSearchIteratorTest {
+
+  private static final SplunkBatchSourceConfig CONFIG = new SplunkBatchSourceConfig(
+    "reference",
+    "basic",
+    "apiToken",
+    "userName",
+    60000,
+    60000,
+    3,
+    60000,
+    100,
+    "invalid",
+    "password",
+    "executionMode",
+    "outputFormat",
+    "searchString",
+    "searchId",
+    0L,
+    "earliestTime",
+    "latestTime",
+    "indexedEarliestTime",
+    "indexedLatestTime",
+    100L,
+    "schema");
+  private static final String SEARCH_ID = "searchId";
+  private static final long OFFSET = 0L;
+
+  @Test
+  public void testSearchIterator() {
+    Map<String, String> expectedEvent1 = ImmutableMap.of("key1", "event1");
+
+    Map<String, String> event1 = ImmutableMap.of("!k^e*y()1{}", "event1");
+    Map<String, String> event2 = ImmutableMap.of("key1", "event2");
+    Map<String, String> event3 = ImmutableMap.of("key1", "event3");
+    List<Event> events = new ArrayList<>();
+    addEvent(events, event1);
+    addEvent(events, event2);
+    addEvent(events, event3);
+
+    InputStream stream = Mockito.mock(InputStream.class);
+    ResultsReader resultsReader = PowerMockito.mock(ResultsReader.class);
+
+    Mockito.when(resultsReader.iterator()).thenReturn(events.iterator());
+
+    SplunkSearchIterator searchIterator = new SplunkSearchIterator(
+      null, CONFIG, SEARCH_ID, OFFSET, SplunkBatchSourceConfig.PARTITION_MAX_SIZE) {
+      @Override
+      ResultsReader getResultsReader(InputStream stream, String outputFormat) {
+        return resultsReader;
+      }
+
+      @Override
+      InputStream getStreamResults(SplunkBatchSourceConfig config,
+                                   String searchId, long offset, Long count) {
+        return stream;
+      }
+    };
+
+    boolean actual = searchIterator.hasNext();
+    Map<String, String> next = searchIterator.next();
+    Assert.assertTrue(actual);
+    Assert.assertEquals(expectedEvent1, next);
+
+    actual = searchIterator.hasNext();
+    next = searchIterator.next();
+    Assert.assertTrue(actual);
+    Assert.assertEquals(event2, next);
+
+    actual = searchIterator.hasNext();
+    next = searchIterator.next();
+    Assert.assertTrue(actual);
+    Assert.assertEquals(event3, next);
+
+    actual = searchIterator.hasNext();
+    Assert.assertFalse(actual);
+
+    Mockito.verify(resultsReader, Mockito.times(1)).iterator();
+  }
+
+  private void addEvent(List events, Map<String, String> event) {
+    events.add(event);
```

Comment:
why do you need this one-liner?

Reply:
`((List) events).add(event1);` vs `addEvent(events, event1);`
Second option looks cleaner.

com.splunk.Event doesn't have public constructor and it is a subclass of HashMap.